File: ai/old/inventory.py
#
# Project: ZAPPY
# File: inventory.py
# Description: functions to interact with the inventory
#

class Inventory:
    def __init__(self, server_connection):
        self.server_connection = server_connection
        self.inventory = {
            'food': 0,
            'linemate': 0,
            'deraumere': 0,
            'sibur': 0,
            'mendiane': 0,
            'phiras': 0,
            'thystame': 0
        }
        self.level = 1

    def get_inventory(self):
        self.server_connection.send_command('Inventory')
        inventory_response = self.server_connection.receive_response()
        self.update_inventory(inventory_response)
        return inventory_response

    def update_inventory(self, inventory_response):
        inventory_response = inventory_response.strip('[]').split(',')
        for item in inventory_response:
            if ' ' in item:
                key, value = item.split()
                self.inventory[key] = int(value)

    def print_inventory(self):
        for resource, quantity in self.inventory.items():
            print(f"{resource}: {quantity}")

File: gui/sources/graphical/components/Inventory.cpp
/*
** EPITECH PROJECT, 2024
** zappy_gui
** File description:
** vim on top !
*/

#include "graphical/components/Inventory.hpp"

using namespace zappy;

void
Inventory::draw()
{
    // Draw the inventory texture
    texture.Draw(position, 0.0f, 2.5f);

    // Draw the resource counts
    int slotHeight = 58;  // Further increase slot height for more spacing
    int textPadding = 10; // Adjust padding from the bottom right corner
    int fontSize = 25;    // Increase font size for better visibility

    vec2 textPos = position;
    textPos.y += 45; // Adjust starting vertical position
    textPos.x -= 38; // Adjust starting horizontal position to move text left

    // Define the correct order of resources
    std::vector<RessourceType> resourceOrder = {
        RessourceType::FOOD,
        RessourceType::LINEMATE,
        RessourceType::DERAUMERE,
        RessourceType::SIBUR,
        RessourceType::MENDIANE,
        RessourceType::PHIRAS,
        RessourceType::THYSTAME
    };

    for (const auto& type : resourceOrder) {
        std::string countText = std::to_string(count[type]);
        int offset = 0;

        if (count[type] > 9 && count[type] != 11) {
            offset = 8;
        }

        raylib::DrawText(countText.c_str(),
                         textPos.x + texture.width * 3 - textPadding - offset,
                         textPos.y - textPadding,
                         fontSize,
                         WHITE);
        textPos.y += slotHeight; // Move to the next slot position
    }
}
